### DISCUSSION TITLE
Use GradientTheme to color PointFeatures

### DISCUSSION THREAD

**Discussion by bor-osbalazs** [Mapsui/Mapsui#1868] [Q&A]
I create a new layer and before creating it I change the color of the point features color.

I'm trying to add style like this:

```
IStyle min = new Style();
IStyle max = new Style();

GradientTheme theme = new GradientTheme("Default", 0, 400, min, max);
theme.FillColorBlend = ColorBlend.Rainbow5;

feature.Styles.Add(theme); 
```

I get the output that I shouldn't use the GradientTheme on individual features, so I tried to add it to the layer when creating it.

```
IStyle min = new Style();
IStyle max = new Style();

GradientTheme theme = new GradientTheme("Default", 0, 400, min, max);
theme.FillColorBlend = ColorBlend.Rainbow5;

var layer = new Layer
{
    Name = "Points",
    DataSource = CreateMemoryProvider(logic),
    Style = theme,
    IsMapInfoLayer = true,
};
```

It still doesn't work, but I don't get any errors. So my question is how can I color circles with gradient from the center? Or create a heatmap like circles?

**Comment by pauldendulk** [comment]
The GradientTheme does not draw a single style with a gradient but draws different styles for different features using gradient like interpolation. Everything you can do with a GradientTheme is also possible with a ThemeStyle.

There is nothing off the shelf in Mapsui to draw gradients in a single feature. It should be possible with a [custom style renderer](https://mapsui.com/documentation/custom-style-renders.html).

To get a heatmap like effect you could also experiment with transparency. When semitransparent points are drawn over each other they can have a similar effect.

I have no experience in generating heatmaps but I could imagine that advanced solutions generate polygons based on the point data and draw those with different colors.

**Reply by bor-osbalazs** [reply]
Thank you for your reply. We use SymbolStyle and not ThemeStyle, but they look the same to me. Is there any difference between these two? Also I was messing around with the custom style renderer, but I can't access the `Mapsui.Geometries` namespace, because they changed to NTS and I can no longer use the Geometry of a feature. Do you have an updated custom style renderer that I could mess around with?

**Reply by pauldendulk** [reply]
The CustomStyleRenderer code sample in the docs is for v3. The current master contains the code sample for v4 https://github.com/Mapsui/Mapsui/blob/master/Samples/Mapsui.Samples.Common/Maps/Styles/CustomStyleSample.cs.
